src/pykeen/hpo/hpo.py

```diff
@@ -353,6 +353,14 @@ def _get_best_study_config(self):
             if field.name.endswith("_kwargs"):
                 logger.debug(f"saving pre-specified field in pipeline config: {field.name}={field_value}")
                 pipeline_config[field.name] = field_value
+            elif field.name == "result_tracker" and field_value:
+                if isinstance(field_value, str):
+                    pipeline_config[field.name] = field_value
+                else:
+                    tracker_subclass = tracker_resolver.normalize_cls(field_value)
+                    if len(tracker_subclass) == 0:  # field_value is base class
+                        continue
+                    pipeline_config[field.name] = tracker_subclass
```

Comment:
```suggestion
                elif issubclass(field_value, ResultTracker):
                    tracker_subclass = tracker_resolver.normalize_cls(field_value)
                    if not tracker_subclass:  # field_value is base class
                        continue
                    pipeline_config[field.name] = tracker_subclass
                else:
                    logger.error(f"Invalid value for field {field.name}: {field_value!r}")
```

While the type annotation of `result_tracker` suggests it can only be an instance or class of `ResultTracker`, you seem to be able to set a string, too. So maybe we should be a bit cautious here, and at least log some error message when something unexpected happens. Usually I would advocate to fail fast, but since this is the code run directly before saving the results of an (expensive) HPO, it might be better to keep on going.

Reply:
(I am not sure where best study config is used; if it is saved to JSON, we may encounter a problem with serialization if the value is something non-JSON-serializable such as an object, or a class)